app/dao/dao_utils.py

```diff
@@ -0,0 +1,34 @@
+from functools import wraps
+
```

Comment:
I do like these decorators.
One scenario which we may have to alter their functionality in the future is to handle situations where we want to commit and situations where it is one of a many db operations and only on the last db operation do we want to commit. How do we handle the stacked decorated methods?